feign-reactor-core/src/main/java/reactivefeign/client/ReactiveHttpRequest.java

```diff
@@ -26,64 +28,82 @@
-                             URI uri, Map<String, List<String>> headers, Publisher<Object> body) {
-    this.methodMetadata = checkNotNull(methodMetadata, "method of %s", uri);
-    this.target = checkNotNull(target, "target of %s", uri);
-    this.uri = checkNotNull(uri, "url");
-    this.headers = checkNotNull(headers, "headers of %s %s", methodMetadata, uri);
-    this.body = body; // nullable
-  }
-
-  public ReactiveHttpRequest(ReactiveHttpRequest request, URI uri) {
-    this(request.methodMetadata, request.target, uri, request.headers, request.body);
-  }
-
-  public ReactiveHttpRequest(ReactiveHttpRequest request, Publisher<Object> body){
-     this(request.methodMetadata, request.target, request.uri, request.headers, body);
-  }
-
-  /* Method to invoke on the server. */
-  public String method() {
-    return methodMetadata.template().method();
-  }
-
-
-
-  /* Fully resolved URL including query. */
-  public URI uri() {
-    return uri;
-  }
-
-  /* Ordered list of headers that will be sent to the server. */
-  public Map<String, List<String>> headers() {
-    return headers;
-  }
-
-  /**
-   * If present, this is the replayable body to send to the server.
-   */
-  public Publisher<Object> body() {
-    return body;
-  }
-
-  public String methodKey(){
-    return methodTag(methodMetadata);
-  }
+    public static final String KEY_VALUE_PAIR_TEMPLATE = "%s=%s";
+    private final MethodMetadata methodMetadata;
+    private final Target<?> target;
+    private final URI uri;
+    private final Map<String, List<String>> headers;
+    private final Publisher<Object> body;
+
+    /**
+     * No parameters can be null except {@code body}. All parameters must be effectively immutable,
+     * via safe copies, not mutating or otherwise.
+     */
+    public ReactiveHttpRequest(MethodMetadata methodMetadata, Target<?> target,
+                               URI uri, Map<String, List<String>> headers, Publisher<Object> body) {
+        this.methodMetadata = checkNotNull(methodMetadata, "method of %s", uri);
+        this.target = checkNotNull(target, "target of %s", uri);
+        this.uri = checkNotNull(uri, "url");
+        this.headers = checkNotNull(headers, "headers of %s %s", methodMetadata, uri);
+        this.body = body; // nullable
+    }
+
+    public ReactiveHttpRequest(ReactiveHttpRequest request, URI uri) {
+        this(request.methodMetadata, request.target, uri, request.headers, request.body);
+    }
+
+    public ReactiveHttpRequest(ReactiveHttpRequest request, Publisher<Object> body) {
+        this(request.methodMetadata, request.target, request.uri, request.headers, body);
+    }
+
+    /* Method to invoke on the server. */
+    public String method() {
+        return methodMetadata.template().method();
+    }
+
+    public ReactiveHttpRequest withQuery(String key, String value) {
+        String query = uri.getQuery();
+        String keyValuePair = String.format(KEY_VALUE_PAIR_TEMPLATE, key, value);
```

Comment:
String.format is extremely ineffective method. Try to avoid it

Reply:
replaced with concatenation